google/auth/external_account_authorized_user.py

```diff
@@ -203,9 +248,7 @@ def refresh(self, request):
             google.auth.exceptions.RefreshError: If the credentials could
                 not be refreshed.
         """
-        if not all(
-            (self._refresh_token, self._token_url, self._client_id, self._client_secret)
-        ):
+        if not self.can_refresh:
             raise exceptions.RefreshError(
                 "The credentials do not contain the necessary fields need to "
                 "refresh the access token. You must specify refresh_token, "
```

Comment:
nit: why not to post only fields that are actually empty?

Reply:
I'm trying to figure out what the error message would look like for that, and I think it would look pretty similar to this one anyways.  It should list out all of the fields that they need, even the ones that they have, and then maybe indicate the fields that are missing from that